Kernel/Time/TimeManagement.cpp

```diff
@@ -142,13 +142,15 @@ UNMAP_AFTER_INIT void TimeManagement::initialize(u32 cpu)
         VERIFY(!s_the.is_initialized());
         s_the.ensure_instance();
 
-        // Initialize the APIC timers after the other timers as the
-        // initialization needs to briefly enable interrupts, which then
-        // would trigger a deadlock trying to get the s_the instance while
-        // creating it.
-        if (auto* apic_timer = APIC::the().initialize_timers(*s_the->m_system_timer)) {
-            dmesgln("Time: Using APIC timer as system timer");
-            s_the->set_system_timer(*apic_timer);
+        if (kernel_command_line().is_ioapic_enabled()) {
```

Comment:
This might be a bit nicer if we just ask the APIC directly if it's available (`APIC::initialized()`) in case we ever add a case in which the user requested the ioapic to be enabled but we decided to not use it anyways. This is fine as is though, so up to you.

Reply:
I was initially tending towards using the option you suggested but went with kernel_command_line switch. But I see your point.